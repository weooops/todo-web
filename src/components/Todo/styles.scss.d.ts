export const todo: string;
export const title: string;
export const message: string;
export const blue: string;
