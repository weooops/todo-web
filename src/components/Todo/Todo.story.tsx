import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { text, withKnobs } from '@storybook/addon-knobs';

import Todo from './Todo';

storiesOf('Todo', module)
  .addDecorator(withKnobs)
  .add('default', () => {
    const props = {
      todo: {
        id: 1,
        title: text('title', 'title'),
        message: text('message', 'message'),
        updatedAt: '2018-01-28 06:52:05'
      }
    };
    return <Todo {...props} />;
  });
