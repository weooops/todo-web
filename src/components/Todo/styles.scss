.todo {
  margin: 10px;
  border-radius: 3px;
  padding: 20px;
  background-color: $dark-gray;
  color: #ffffff;

  .title {
    margin-bottom: 10px;
    font-size: $normal-font-size + 10;
    font-weight: lighter;
    word-wrap: break-word;
  }

  .message {
    margin-bottom: 10px;
    line-height: 1.25;
    word-wrap: break-word;
  }

  .updatedAt {
    font-size: 12px;
  }
}

