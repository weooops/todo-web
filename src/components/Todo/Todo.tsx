import * as React from 'react';
import * as distanceInWordsToNow from 'date-fns/distance_in_words';
import * as ko from 'date-fns/locale/ko';

import * as styles from './styles.scss';
import * as classNames from 'classnames/bind';
import { TodoType } from '../../models';

const cx = classNames.bind(styles);

interface IProps {
  todo: TodoType;
}

class Todo extends React.Component<IProps> {
  colors = [
    '#EF5350', '#EC407A', '#AB47BC', '#7E57C2', '#5C6BC0',
    '#42A5F5', '#29B6F6', '#26C6DA', '#26A69A', '#66BB6A',
    '#9CCC65', '#D4E157', '#FFEE58', '#FFCA28', '#FFA726',
    '#FF7043', '#8D6E63', '#BDBDBD'
  ];

  public shouldComponentUpdate(nextProps: IProps): boolean {
    return nextProps.todo !== this.props.todo;
  }

  public render(): JSX.Element {
    const { todo: { title, message, updatedAt } } = this.props;
    const pos = Math.round(Math.random() * this.colors.length - 1);

    return (
      <div className={cx('todo')} style={{backgroundColor: this.colors[pos]}}>
        <div className={cx('title')}>{title}</div>
        <div className={cx('message')}>{message}</div>
        <div className={cx('updatedAt')}>{distanceInWordsToNow(updatedAt, new Date(), { locale: ko })}</div>
      </div>
    );
  }
}

export default Todo;
