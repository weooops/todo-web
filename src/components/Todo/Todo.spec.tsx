import * as React from 'react';
import { shallow } from 'enzyme';

import Todo from './Todo';

it('renders correctly', () => {
  const props = {
    todo: {
      id: 1,
      title: 'title',
      message: 'message',
      updatedAt: '2018-01-28 06:52:05'
    }
  };
  const wrapper = shallow(<Todo {...props} />);
  expect(wrapper).toMatchSnapshot();
});
