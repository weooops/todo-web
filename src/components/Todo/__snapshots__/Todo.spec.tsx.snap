// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    todo={
      Object {
        "id": 1,
        "message": "message",
        "title": "title",
        "updatedAt": "2018-01-28 06:52:05",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          title
        </div>,
        <div
          className="message"
        >
          message
        </div>,
        <div
          className="updatedAt"
        >
          1일
        </div>,
      ],
      "className": "todo",
      "style": Object {
        "backgroundColor": "#9CCC65",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "className": "title",
        },
        "ref": null,
        "rendered": "title",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "message",
          "className": "message",
        },
        "ref": null,
        "rendered": "message",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1일",
          "className": "updatedAt",
        },
        "ref": null,
        "rendered": "1일",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            title
          </div>,
          <div
            className="message"
          >
            message
          </div>,
          <div
            className="updatedAt"
          >
            1일
          </div>,
        ],
        "className": "todo",
        "style": Object {
          "backgroundColor": "#9CCC65",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "title",
          },
          "ref": null,
          "rendered": "title",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "message",
            "className": "message",
          },
          "ref": null,
          "rendered": "message",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1일",
            "className": "updatedAt",
          },
          "ref": null,
          "rendered": "1일",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
